src/root/real_t.d

```diff
@@ -0,0 +1,118 @@
+// Compiler implementation of the D programming language
+// Copyright (c) 1999-2016 by Digital Mars
+// All Rights Reserved
+// written by Walter Bright
+// http://www.digitalmars.com
+// Distributed under the Boost Software License, Version 1.0.
+// http://www.boost.org/LICENSE_1_0.txt
+
+module ddmd.root.real_t;
+
+import core.stdc.errno;
+import core.stdc.stdio;
+import core.stdc.stdlib;
+import core.stdc.string;
+
+alias real_t = real;
+
+private
+{
+    version(CRuntime_DigitalMars) __gshared extern (C) extern const(char)* __locale_decpoint;
+
+    version(CRuntime_Microsoft) extern (C++)
+    {
+        struct longdouble { real_t r; }
+        size_t ld_sprint(char* str, int fmt, longdouble x);
+        longdouble strtold_dm(const(char)* p, char** endp);
+    }
+}
+
+extern (C++, target_real)
+{
+    __gshared real_t snan;
+
+    real_t sqrt(real_t x)
+    {
+        import core.math;
+        return core.math.sqrt(x);
+    }
+
+    bool areBitwiseEqual(real_t a, real_t b)
+    {
+        return memcmp(&a, &b, 10) == 0;
+    }
+
+    bool isNaN(real_t r)
+    {
+        return !(r == r);
+    }
+
+    bool isSignallingNaN(real_t r)
+    {
+        return isNaN(r) && !(((cast(ubyte*)&r)[7]) & 0x40);
+    }
+
+    bool isInfinity(real_t r)
+    {
+        return r is real_t.infinity || r is -real_t.infinity;
+    }
+
+    real_t parseTargetReal(const(char)* literal, bool* isOutOfRange = null)
+    {
+        errno = 0;
+        version(CRuntime_DigitalMars)
+        {
+            auto save = __locale_decpoint;
+            __locale_decpoint = ".";
+        }
+        version(CRuntime_Microsoft)
+            auto r = strtold_dm(literal, null).r;
+        else
+            auto r = strtold(literal, null);
+        version(CRuntime_DigitalMars) __locale_decpoint = save;
+        if (isOutOfRange)
+            *isOutOfRange = (errno == ERANGE);
+        return r;
+    }
+
+    int printTargetReal(char* str, char fmt, real_t x)
```

Comment:
This should be 'sprintXXX' to suggest that output is written to buffer, and not stdout.